### PR TITLE
oxigraph: init at 0.3.11

### PR DESCRIPTION
**Pull request by astro**
###### Description of changes

There are associated Python libraries to be packaged in the future. Yet the server is usable on its own through HTTP APIs.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/servers/oxigraph/default.nix [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+  # requires packaging of the associated python modules
+  doCheck = false;
+
+  meta = with lib; {
+    homepage = "https://github.com/oxigraph/oxigraph";
+    description = "SPARQL graph database";
+    platforms = platforms.unix;
+    maintainers = with maintainers; [ astro ];
+    licenses = with licenses; [ asl20 mit ];
```
Comment (kilianar) [inline pkgs/servers/oxigraph/default.nix:42]:
```suggestion
    license = with licenses; [ asl20 mit ];
```

File: pkgs/servers/oxigraph/default.nix [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    sha256 = "sha256-7KbDZKKJPk3QTp4siIbdB6xKbslw73Lhc7NoeOuA0Og=";
+    fetchSubmodules = true;
+  };
+
+  cargoSha256 = "sha256-Yqn6hwejg6LzcqW0MiUN3tqrOql6cpu/5plaOz+2/ns=";
+
+  nativeBuildInputs = [
+    pkg-config llvmPackages.clang
+  ];
+  LIBCLANG_PATH = "${llvmPackages.libclang.lib}/lib";
```
Comment (NickCao) [inline pkgs/servers/oxigraph/default.nix:25]: There is `rustPlatform.bindgenHook``

Comment (astro) [reply pkgs/servers/oxigraph/default.nix]: Thanks!

**Comment by NickCao** [comment]
build is failing on darwin.

**Comment by astro** [comment]
> build is failing on darwin.

It is trying to run rustfmt on that platform. Just why?

**Comment by NickCao** [comment]
Lacking rustfmt is non-fatal, the log suggests, and it's the compilation of rocksdb failing. You can try using system roscksdb.

**Comment by NickCao** [comment]
`note: ld: framework not found Security`

**Comment by NickCao** [comment]
still broken on x86_64-darwin.

**Comment by NickCao** [comment]
And squash the commits?

**Comment by astro** [comment]
Squashed!

File: pkgs/servers/oxigraph/default.nix [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+    sha256 = "sha256-7KbDZKKJPk3QTp4siIbdB6xKbslw73Lhc7NoeOuA0Og=";
+    fetchSubmodules = true;
+  };
+
+  cargoSha256 = "sha256-Yqn6hwejg6LzcqW0MiUN3tqrOql6cpu/5plaOz+2/ns=";
+
+  nativeBuildInputs = [
+    pkg-config
+    rustPlatform.bindgenHook
+  ] ++ lib.optional stdenv.isDarwin rustfmt;
```
Comment (NickCao) [inline pkgs/servers/oxigraph/default.nix:28]: Just remove rustfmt, It's optional (and does no good other than suppressing a few lines of harmless build log).

File: pkgs/servers/oxigraph/default.nix [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "sha256-7KbDZKKJPk3QTp4siIbdB6xKbslw73Lhc7NoeOuA0Og=";
+    fetchSubmodules = true;
+  };
+
+  cargoSha256 = "sha256-Yqn6hwejg6LzcqW0MiUN3tqrOql6cpu/5plaOz+2/ns=";
+
+  nativeBuildInputs = [
+    pkg-config
```
Comment (NickCao) [inline pkgs/servers/oxigraph/default.nix:26]: There are no buildInputs on linux, and maybe darwin does not use pkg-config to locate frameworks?

File: pkgs/servers/oxigraph/default.nix [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+    rustPlatform.bindgenHook
+  ] ++ lib.optional stdenv.isDarwin rustfmt;
+  buildInputs = lib.optionals stdenv.isDarwin [ IOKit Security ];
+
+  preConfigure = ''
+    cd server
+  '';
+  postBuild = ''
+    cd ..
+  '';
```
Comment (NickCao) [inline pkgs/servers/oxigraph/default.nix:36]: does `cargoBuildFlags = [ "--package" "server" ]` achieve the same effect?

Comment (astro) [reply pkgs/servers/oxigraph/default.nix:36]:
```
error: package ID specification `server` did not match any packages
```

Comment (NickCao) [reply pkgs/servers/oxigraph/default.nix:36]: Or maybe `oxigraph_server`?

File: pkgs/servers/oxigraph/default.nix (hunk 3 above)
Comment (astro) [reply pkgs/servers/oxigraph/default.nix]: ok

File: pkgs/servers/oxigraph/default.nix (hunk 5 above)
Comment (astro) [reply pkgs/servers/oxigraph/default.nix]: That did the trick!

**Review by NickCao — APPROVED** [review]

**Pull request merged**